### ISSUE TITLE
CUB_BIRD 5shot could acheieve 82% acc, use pretrained resnet18 on Imagenet?

### ISSUE THREAD

**Issue by xhw205** [tankche1/Semantic-Feature-Augmentation-in-Few-shot-Learning#4]

**Comment by xhw205** [comment]
I use the rs 18 as feature extrator+SVM classifier，not pre-trained on Imagenet， only 50% on CUBBIRDS datasets。

**Comment by tankche1** [comment]
I'm quite sure that you have made some mistakes.ResNet18 trained from scratch on CUB can achieve 82%.

**Comment by xhw205** [comment]
1.Train a new rs18-model， only use 100 class in CUBBirds？
2.and could you contribute this part of the the code...I couldn't achieve 82% acc as you say in paper

**Comment by tankche1** [comment]
1.Yes.
2.You only need to write a new dataloader for CUB. I used the same code for all datasets.
3.You can observe similar results in some papers.50% is impossible.
∆-encoder - an effective sample synthesis method for few-shot object recognition
Deep Meta-Learning Learning to Learn in the Concept Space

**Comment by kailigo** [comment]
I am unable to reproduce 82% acc by training from scatch and use the provided split. Similarly, I got only about 50%. I organize cub dataset in the same way as mini-imagenet and use the same data loader. I also changed the embedding dimension from 64 to 100 to match the case of cub. I guess there are other  modifications to be made to reproduce your result. Could you upload your baseline code (resent18+SVM) for cub dataset, or double-check other adaptation you made to migrate from mini-imagenet to CUB. Your help is much apppreciated

**Comment by xhw205** [comment]
Me too，I use datasets.ImageFolder function to create a CUB dataloader 100 class train rs18 from scratch ，SVM  50  test class 。only 60+%

**Comment by kailigo** [comment]
@tankche1 , after some changes, I can now reach 54.6% for 1-shot and 75.3% for the 5-shot. But it is still much worse than the reported 66.5% and 82.38%. Could you help provide some suggestions for improve my results.  Thanks for your help.

**Comment by kailigo** [comment]
@tankche1 , could you help upload your resnet18 model for the cub dataset, just like you did for the mini-imagenet dataset. I am working on building some new stuff on your work and must first ensure having replicated your results. Thanks.

**Comment by tankche1** [comment]
@kailigo ,I do the experiments a few months ago so i can only give you some tips. Some of my friends reproduce the results easily.
1.In SVM stage,i don't use any trick.(Although there are some technics to do so.).Maybe you can try Logistic Regression (C=10) to check if something went wrong in this stage.
2.In classification stages, just normal classification task.  You should try to adapt some parameters(LR,SGD or Adam,weight decay,batch size) to make the best feature extractor.
I guess the reason might be the parameters in classification. I'm sure that ResNet18 with CSEloss without rings and bells can achieve this result.



Good Luck!

**Comment by kailigo** [comment]
@tankche1 thanks for the response. I will tune the feature extractor. By the way, could you do the favor and  take a look of your code and provide recommended parameter settings. This will tremendously reduce the trials required to achieve the reported results. Thanks again for your help.

**Comment by kailigo** [comment]
@tankche1 . I tried various parameter settings to get a better feature extractor, but it does not seems help. I trained resnet18 with 200 and 300 epochs (the defaulted one is 100) and tuned the learning rate from 1e-3 to 1e-2. These changes bring me an increase of the training accuracy from ~67% to ~77%. It seemed that a better feature extraction network has been obtained. However, as I test it for few-shot classification, the accuracy surprisingly turns even lower to ~51% (previous it was 54%). I wonder if you can provide some comments on this fact and any suggestions to improve the results. Thanks

**Comment by kailigo** [comment]
Any help? Still waiting +_+

**Comment by tankche1** [comment]
@kailigo  
[CUB related code.zip](https://github.com/tankche1/Semantic-Feature-Augmentation-in-Few-shot-Learning/files/2290281/CUB.related.code.zip)

The models are too large.

**Comment by tankche1** [comment]
@kailigo I update the git with the CUB model.

**Comment by tankche1** [comment]
@kailigo I update the git with the CUB model.

**Comment by kailigo** [comment]
@tankche1 , much appreciated for your update. I will try it right now and feed back you soon. Cheers.

**Comment by kailigo** [comment]
@tankche1 , I tested your provided model, got 61.7% for 1-shot and 82% for 5-shots. For 5-shots, it matches exactly the reported, but for 1-shot, about 5 points lower than expected. How about your result for the 1-shot case?

**Comment by kailigo** [comment]
@tankche1 , may i ask if you have tested your model for the 1-shot case. Could you help upload the model which you use to get the reported 1-shot results. Thanks.
